src/wireviz/build_examples.py

```diff
@@ -1,113 +1,149 @@
-            if fnmatch(fn, "ex*.yml"):
-                i = ''.join(filter(str.isdigit, fn))
-
-                abspath = os.path.join(examples_path, fn)
-                outfile_name = abspath.split(".yml")[0]
-
-
-                print(abspath)
-                wireviz.parse_file(abspath)
-
-                file.write(f'## Example {i}\n')
-                file.write(f'![]({outfile_name}.png)\n\n')
-                file.write(f'[Source]({fn}) - [Bill of Materials]({outfile_name}.bom.tsv)\n\n\n')
-
-def build_tutorials():
-    with open_file_write(os.path.join(tutorials_path, readme)) as file:
-        file.write('# WireViz Tutorial\n')
-        for fn in sorted(os.listdir(tutorials_path)):
-            if fnmatch(fn, "tutorial*.yml"):
-                i = ''.join(filter(str.isdigit, fn))
-                abspath = os.path.join(tutorials_path, fn)
-                print(abspath)
-
-                wireviz.parse_file(abspath)
-
-                outfile_name = abspath.split(".yml")[0]
-
-                with open_file_read(outfile_name + '.md') as info:
-                    for line in info:
-                        file.write(line.replace('## ', '## {} - '.format(i)))
-                file.write(f'\n[Source]({fn}):\n\n')
-
-                with open_file_read(abspath) as src:
-                    file.write('```yaml\n')
-                    for line in src:
-                        file.write(line)
-                    file.write('```\n')
-                file.write('\n')
-
-                file.write('\nOutput:\n\n'.format(i))
-
-                file.write(f'![](tutorial{outfile_name}.png)\n\n')
-
-                file.write(f'[Bill of Materials](tutorial{outfile_name}.bom.tsv)\n\n\n')
-
-def clean_examples():
-    generated_extensions = ['.gv', '.png', '.svg', '.html', '.bom.tsv']
-
-    for filepath in [examples_path, demos_path, tutorials_path]:
-        print(filepath)
-        for file in sorted(os.listdir(filepath)):
-            if os.path.exists(os.path.join(filepath, file)):
-                if list(filter(file.endswith, generated_extensions)) or file == 'readme.md':
-                    print('rm ' + os.path.join(filepath, file))
-                    os.remove(os.path.join(filepath, file))
+groups = {
+    'examples': {
+        'path': Path(script_path).parent.parent.parent / 'examples',
+        'prefix': 'ex',
+        readme: [], # Include no files
+        'title': 'Example Gallery',
+    },
+    'tutorial' : {
+        'path': Path(script_path).parent.parent.parent / 'tutorial',
+        'prefix': 'tutorial',
+        readme: ['md', 'yml'], # Include .md and .yml files
+        'title': 'WireViz Tutorial',
+    },
+    'demos' : {
+        'path': Path(script_path).parent.parent.parent / 'examples',
+        'prefix': 'demo',
+    },
+}
+
+input_extensions = ['.yml']
+generated_extensions = ['.gv', '.png', '.svg', '.html', '.bom.tsv']
+extensions_not_from_graphviz = [ext for ext in generated_extensions if ext[-1] == 'v']
```

Comment:
This line is a bit cryptic (or maybe I'm just tired)... could you explain the logic? Thanks!

Reply:
In the interest of readability, maybe listing the `extensions_not_from_graphviz` explicitly is better than relying on the letter v ;-)

The fact that `.gv` is marked as "not from GraphViz" is ironic, but understandable.

Maybe shuffle the logic around a bit and have
```python
graphviz_generated_extensions = ['.png', '.svg', '.html']`
other_extensions = ['.gv', '.bom.tsv']  # or some better variable name
```
?

Reply:
Sorry, I forgot to put a comment here about this temporary hack. I collect all `generated_extensions` that ends with a 'v' to filter out the Graphviz output, but I can do it in a more readable way instead.